### PR TITLE
[Broker] Fix call sync method in async rest api for internalTerminatePartitionedTopic

### PR DESCRIPTION
**Pull request by liudezhi2098**
### Motivation

Avoid call sync method in async rest API for PersistentTopicsBase#internalTerminatePartitionedTopic.

### Modifications

- *Use async instead of sync method.*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

Need to update docs? 
  
- [x] `no-need-doc`

### CODE REVIEW

**Review by mattisonchao — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -3264,55 +3264,73 @@ protected MessageId internalTerminate(boolean authoritative) {
...
+                                    log.error("[{}] Failed to terminate topic {}", clientAppId(), topicName, t);
+                                    asyncResponse.resume(new RestException(t));
+                                }
+                            }
+                            asyncResponse.resume(messageIds);
+                            return null;
+                        });
+                    }
+                }).exceptionally(e -> {
+                    Throwable cause = e.getCause();
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:3323]: Can this exception be handled by the `exceptionally` outside?
Seems like duplicate code.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 2]
```diff
@@ -3264,55 +3264,73 @@ protected MessageId internalTerminate(boolean authoritative) {
...
-                    + "not allowed using partitioned-terminate, please use terminate commands.");
-        }
-        if (partitionMetadata.partitions > 0) {
-          final List<CompletableFuture<MessageId>> futures = Lists.newArrayList();
+            future = validateGlobalNamespaceOwnershipAsync(namespaceName);
+        } else {
+            future = CompletableFuture.completedFuture(null);
+        }
+
+        future.thenAccept(__ -> validateTopicOwnershipAsync(topicName, authoritative)
```
Comment (gaozhangmin) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:3274]: Maybe here, We should first check the partition num before validateTopicOwnershipAsync. if partitionMetadata.partitions == 0, There is no need to validateTopicOwnershipAsync

**Review by gaozhangmin — CHANGES_REQUESTED** [review]

**Review by gaozhangmin — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 3]
```diff
@@ -3264,55 +3264,71 @@ protected MessageId internalTerminate(boolean authoritative) {
...
 
-      PartitionedTopicMetadata partitionMetadata = getPartitionedTopicMetadata(topicName, authoritative, false);
-
-        if (partitionMetadata.partitions == 0) {
-            throw new RestException(Status.METHOD_NOT_ALLOWED, "Termination of a non-partitioned topic is "
-                    + "not allowed using partitioned-terminate, please use terminate commands.");
-        }
-        if (partitionMetadata.partitions > 0) {
-          final List<CompletableFuture<MessageId>> futures = Lists.newArrayList();
+        future.thenAccept(__ -> validateTopicOperationAsync(topicName, TopicOperation.TERMINATE)
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:3274]: need to use `thenCompose` to make it make check take effect.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 4]
```diff
@@ -3264,55 +3264,71 @@ protected MessageId internalTerminate(boolean authoritative) {
...
+                .thenAccept(partitionMetadata -> {
+                    if (partitionMetadata.partitions == 0) {
+                        String msg = "Termination of a non-partitioned topic is not allowed using partitioned-terminate"
+                                + ", please use terminate commands";
+                        log.error("[{}] [{}] {}", clientAppId(), topicName, msg);
+                        asyncResponse.resume(new RestException(Status.METHOD_NOT_ALLOWED, msg));
+                        return;
+                    }
+                    if (partitionMetadata.partitions > 0) {
+                        validateTopicOwnershipAsync(topicName, authoritative).thenAccept(unused -> {
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:3285]: 1. exception of `validateTopicOwnershipAsync` seems not handled.
2. I think we don't need to add `validateTopicOwnershipAsync` here.

**Review by Jason918 — CHANGES_REQUESTED** [review]

**Review by Jason918 — APPROVED** [review]

**Pull request merged**